File: conectopia/app/models/Member.scala
package models


import javax.persistence._

@Entity
@Table(name = "member")
class Member extends PersistentEntity {
  // role_id

  @OneToOne
  @JoinColumn(name = "role_id")
  var role: Role = _
  // server_id
  @ManyToOne
  @JoinColumn(name = "server_id")
  var server: Server = _
  // user_id
  @ManyToOne
  @JoinColumn(name = "user_id")
  var user: User = _

  // constructor
  def this(role: Role, server: Server, user: User) {
    this()
    this.role = role
    this.server = server
    this.user = user
  }
}

File: play-1.5.3/Conectopia/app/models/Member.java
package models;

import javax.persistence.Entity;
import play.db.jpa.Model;

import javax.persistence.ManyToOne;

@Entity
public class Member extends Model {
    @ManyToOne
    public Role role;

    @ManyToOne
    public User user;
    @ManyToOne
    public Server server;

    public Member(Role role, User user, Server server) {
        this.role = role;
        this.user = user;
        this.server = server;
    }

    public Member() {
        this.role = null;
        this.user = null;
        this.server = null;
    }
}
